rasa/core/validator.py

```diff
@@ -0,0 +1,178 @@
+
+        domain_intents = []
+        nlu_data_intents = []
+
+        for intent in self.domain.intent_properties:
+            domain_intents.append(intent)
+
+        for intent in self.intents.intent_examples:
+            nlu_data_intents.append(intent.data["intent"])
+
+        for intent in domain_intents:
+            found = self._search(nlu_data_intents, intent)
+            if not found:
+                logger.error(
+                    "The intent '{}' is listed in the domain file, but "
+                    "is not found in the NLU training data.".format(intent)
+                )
+            else:
+                self.valid_intents.append(intent)
+
+        for intent in nlu_data_intents:
+            found = self._search(domain_intents, intent)
+            if not found:
+                logger.error(
+                    "The intent '{}' is in the NLU training data, but "
+                    "is not listed in the domain.".format(intent)
+                )
+
+    def verify_intents_in_stories(self):
+        """Checks intents used in stories.
+
+        Verifies if the intents used in the stories are valid, and whether
+        all valid intents are used in the stories."""
+
+        if self.valid_intents == []:
+            self.verify_intents()
+
+        stories_intents = []
+        for story in self.stories:
+            for event in story.events:
+                if type(event) == UserUttered:
+                    intent = event.intent["name"]
+                    stories_intents.append(intent)
+                    found = self._search(self.valid_intents, intent)
+
+                    if not found:
+                        logger.error(
+                            "The intent '{}' is used in stories, but is not a "
+                            "valid intent.".format(intent)
+                        )
+
+        for intent in self.valid_intents:
+            found = self._search(stories_intents, intent)
+            if not found:
+                logger.warning(
+                    "The intent '{}' is not used in any story.".format(intent)
+                )
+
+    def verify_utterances(self):
+        """Compares list of utterances in actions with utterances in templates."""
+
+        actions = self.domain.action_names
+        utterance_templates = []
+
+        for utterance in self.domain.templates:
+            utterance_templates.append(utterance)
+
+        for utterance in utterance_templates:
+            found = self._search(actions, utterance)
+            if not found:
+                logger.error(
+                    "The utterance '{}' is not listed under 'actions' in the domain file.".format(
+                        utterance
+                    )
+                )
+            else:
+                self.valid_utterances.append(utterance)
+
+        for action in actions:
+            if action.split("_")[0] == "utter":
```

Comment:
We have a constant for this, it'll look something like "if action.startswith(UTTER_PREFIX)", you can import it from rasa.constants